feat/keyword-read

backswan0 opened the pull request:
## #️⃣ Kan Board Number

7, 20, 21

## 📝 요약
모든 키워드 조회 (R) 구현 - Postman 테스트 완료
사용자가 알림으로 등록한 채용 키워드 목록 조회 (R) 구현 완료 - Postman 테스트 완료
조회할 때 키워드의 id와 이름을 함께 반환하도록 리팩토링 완료 


## 🛠️ PR 유형

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 💬 논의사항 or 질문

현재 사용자가 알림 받을 키워드를 등록할 때에는,
{
    "keywordIdList": [1, 3]
}
위와 같이 입력하고,
{
      "keywordList": [
          "Java",
          "Kafka"
      ]
}
응답은 예시처럼 이름만 나오는데요. 여기서 응답할 때에도 id를 함께 전달해줘야 할지 의견을 구하고 싶습니다:)

## ✅ PR Checklist

PR이 다음 요구 사항을 충족했는지 확인하기!

- [x] 커밋 메시지 컨벤션 준수
- [x] 변경 사항 테스트 여부 체크(버그 수정/기능 테스트)

## 💥 리뷰어 지목!!!!!

<!--- 리뷰어를 2명 이상 지목해주세요. -->
- [x] 채영
- [ ] 지현
- [x] 리은
- [ ] 승찬
- [x] 주양

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - 전체 키워드 목록을 조회할 수 있는 새로운 API 엔드포인트가 추가되었습니다.
  - 사용자 별로 키워드 정보를 확인할 수 있는 기능이 도입되었습니다.
  - 향상된 응답 구조를 통해 키워드 데이터 전달이 개선되어 사용자 경험이 향상되었습니다.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

backswan0 committed:
:sparkles: feat: 모든 키워드 조회 (R) 구현

- 사용자가 알림으로 등록할 수 있는 키워드 목록을 조회하는 API 구현

backswan0 committed:
:sparkles: feat: 사용자 알림 채용 키워드 조회 (R) 구현

- 사용자가 본인이 어떤 키워드를 알림으로 등록했는지 조회하는 기능 구현

- [할 일] id 함께 반환하기

backswan0 committed:
:recycle: refactor: 모든 키워드 조회 시 키워드의 아이디와 이름을 함께 반환하도록 리팩토링

- 키워드 조회 시 키워드의 아이디와 이름을 함께 반환하도록 리팩토링

backswan0 committed:
:recycle: refactor: 사용자 알림 채용 키워드 조회 (R) 리팩토링

- 사용자가 알림으로 등록한 키워드 조회 시 키워드의 아이디와 이름을 함께 반환하도록 리팩토링

- [할 일] 머지 이후 사용자 알림 채용 키워드 등록 (C) 또한 위와 같은 로직으로 리팩토링

juyangjin commented on src/main/java/com/project/cheerha/domain/keyword/dto/response/ReadKeywordResponseDto.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.List;
+
+public record ReadKeywordResponseDto(List<KeywordDto> KeywordList) {
+    // 응답 화면에서 'KeywordList'로 나오도록 'KeywordDtoList' 대신 'KeywordList'로 명명
```
제 생각에 응답화면을 이렇게까지 고려해서 만드는 게 맞는가? 하는 의문이 듭니다. 
응답 화면에 대한 건은 사실 우리가 정보를 프론트에 넘겨준 후에 프론트 쪽에서 정리해야 할 문제가 아닐까 합니다. 

지금은 이렇게 한두개이긴 하지만 이런 예외사항을 두면서 네이밍 규칙과 다르게 작성되면 나중에 유지보수 할때 좀 곤란해질 거 같다는 생각도 들고요.

backswan0 replied:
수정했습니다:)

juyangjin commented on src/main/java/com/project/cheerha/domain/keyword/dto/response/ReadUserKeywordResponseDto.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.List;
+
+public record ReadUserKeywordResponseDto(List<KeywordDto> keywordListChosenByUser) {
+    // 응답 화면에서 'keywordListChosenByUser'로 나오도록 위와 같이 변수 명명
```
이부분도 마찬가지로 프론트 쪽에서 신경쓸 문제라고 봅니다.

backswan0 replied:
수정했습니다:)

juyangjin commented on src/main/java/com/project/cheerha/domain/keyword/service/UserKeywordService.java:
```diff
@@ -96,4 +98,27 @@ public void deleteUserKeyword(
+                    );
+                }
+            )
+            .toList();
```
107번 줄부터 여기까지 코드가 가독성이 좀 떨어지는 듯한 느낌이 듭니다.  이 부분은 조금 더 간결하게 리팩토링을 해도 좋을 거 같다는 생각이 듭니다.

backswan0 replied:
수정했습니다:)

backswan0 committed:
Merge remote-tracking branch 'origin/dev' into feat/keyword-read

backswan0 committed:
:memo: chore: 변수 이름 수정 및 코드 포맷팅

- 변수명 팀 코드 컨벤션에 맞추어서 수정

- 가독성을 높이고자 getKeywordById 및 getUserById 메서드를 private 메서드로 추출

nbcamp-spring commented:
@coderabbitai review
